app/src/main/java/com/example/touchcar/presentation/choose_model/ChooseModelFragment.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.touchcar.presentation.choose_model
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import androidx.recyclerview.widget.RecyclerView
+import com.example.touchcar.R
+import com.example.touchcar.presentation.choose_model.recycler.ChooseModelAdapter
+import com.example.touchcar.presentation.utils.addTextChangedListener
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class ChooseModelFragment : Fragment() {
+
+    @Inject
+    lateinit var chooseModelViewModel: ChooseModelViewModel
+    lateinit var modelUrl: String
+    private lateinit var chooseModelAdapter: ChooseModelAdapter
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var searchBar: EditText
+
+    companion object {
+        fun newInstance() = ChooseModelFragment()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.choose_model_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        modelUrl = arguments?.get("url") as String
+        chooseModelViewModel.modelLiveData
+            .observe(
+                this, { models ->
```

Comment:
третий раз про это пишу) у тебя у самого глаза не выпадают здесь?
зацени, насколько читаемее такой вариант
```suggestion
        chooseModelViewModel.modelLiveData.observe(this) { models ->
```